src/nvim/eval/funcs.c

```diff
@@ -10785,21 +10785,30 @@ static void f_termopen(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   }
 
   // Terminal URI: "term://$CWD//$PID:$CMD"
-  snprintf((char *)NameBuff, sizeof(NameBuff), "term://%s//%d:%s",
-           (char *)IObuff, pid, cmd);
+  buf_name_for_term(NameBuff, sizeof(NameBuff), IObuff, pid, cmd);
   // at this point the buffer has no terminal instance associated yet, so unset
   // the 'swapfile' option to ensure no swap file will be created
   curbuf->b_p_swf = false;
   (void)setfname(curbuf, NameBuff, NULL, true);
-  // Save the job id and pid in b:terminal_job_{id,pid}
+  // Save the job id, pid, command and cwd in b:terminal_job_{id,pid,cmd,cwd}
   Error err = ERROR_INIT;
   // deprecated: use 'channel' buffer option
-  dict_set_var(curbuf->b_vars, cstr_as_string("terminal_job_id"),
+  dict_set_var(curbuf->b_vars, STATIC_CSTR_AS_STRING("terminal_job_id"),
```

Comment:
these are deprecated (see above comment). instead the 'channel' buffer-local option holds the channel-id (job-id).